### PR TITLE
Veto loading a project with missing capabilities

### PR DESCRIPTION
**Pull request by davkean**
This veto loads a project with missing capabilities to catch situations such as https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1083914/.

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/ProjectCapabilitiesMissingVetoProjectLoad.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        private readonly IProjectCapabilitiesService _projectCapabilitiesService;
+
+        [ImportingConstructor]
+        public ProjectCapabilitiesMissingVetoProjectLoad(UnconfiguredProject project, IProjectCapabilitiesService projectCapabilitiesService)
+        {
+            _project = project;
+            _projectCapabilitiesService = projectCapabilitiesService;
+        }
+
+        public Task<bool> AllowProjectLoadAsync(bool isNewProject, ProjectConfiguration activeConfiguration, CancellationToken cancellationToken = default)
```
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/ProjectCapabilitiesMissingVetoProjectLoad.cs]: This will never return false.

Comment (davkean) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/ProjectCapabilitiesMissingVetoProjectLoad.cs]: Yes.

Comment (drewnoakes) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/ProjectCapabilitiesMissingVetoProjectLoad.cs]: Never mind. I see the interface's docs.

Comment (drewnoakes) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/ProjectCapabilitiesMissingVetoProjectLoad.cs]: I didn't realise this implemented an interface. I was hinting at changing the return type.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/ProjectCapabilitiesMissingVetoProjectLoad.ProjectType.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS
+{
+    internal partial class ProjectCapabilitiesMissingVetoProjectLoad
+    {
+        private class ProjectType
```
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/ProjectCapabilitiesMissingVetoProjectLoad.ProjectType.cs]: Could be a `readonly struct`.

Comment (davkean) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/ProjectCapabilitiesMissingVetoProjectLoad.ProjectType.cs]: Could be but would result in basically unneeded JIT code gen for basically 3 instances on the heap.

**Pull request merged**